Tests/AppcuesKitTests/Analytics/AnalyticsTrackerTests.swift

```diff
@@ -66,6 +66,123 @@ class AnalyticsTrackerTests: XCTestCase {
+        appcues.activityProcessor.onProcess = { activity, completion in
+            ["userProp":1].verifyPropertiesMatch(activity.profileUpdate)
+            ["groupProp":2].verifyPropertiesMatch(activity.groupUpdate)
+            XCTAssertEqual("test-group", activity.groupID)
+            onRequestExpectation.fulfill()
+        }
+
+        let profileUpdate = TrackingUpdate(type: .profile(interactive: true), properties: ["userProp":1], isInternal: false)
+        let groupUpdate = TrackingUpdate(type: .group("test-group"), properties: ["groupProp":2], isInternal: false)
+
+        // Act
+        tracker.track(update: profileUpdate)
+        appcues.storage.groupID = "test-group" // mimick what is done in Appcues group() call
+        tracker.track(update: groupUpdate)
+
+        // Assert
+        waitForExpectations(timeout: 1)
+    }
+
+    func testIdentifyWithDifferentUser() throws {
+        // Arrange
+        let onRequestExpectation = expectation(description: "Valid request")
+        onRequestExpectation.expectedFulfillmentCount = 2
+
+        var updateCount = 0
+
+        // In the case of a profile update, immediately followed by another identify with a different
+        // user, there should be no batching of the Activity, as the first user profile update must
+        // be sent, then the second update with the new user.
+        appcues.activityProcessor.onProcess = { activity, completion in
+            updateCount += 1
+            if updateCount == 1 {
+                ["userProp":1].verifyPropertiesMatch(activity.profileUpdate)
+                XCTAssertEqual("user-1", activity.userID)
+            } else if updateCount == 2 {
+                ["userProp":2].verifyPropertiesMatch(activity.profileUpdate)
+                XCTAssertEqual("user-2", activity.userID)
+            }
+            onRequestExpectation.fulfill()
+        }
+
+        let profileUpdate1 = TrackingUpdate(type: .profile(interactive: true), properties: ["userProp":1], isInternal: false)
+        let profileUpdate2 = TrackingUpdate(type: .profile(interactive: true), properties: ["userProp":2], isInternal: false)
+
+        // Act
+        appcues.storage.userID = "user-1" // mimick what is done in Appcues identify() call
+        tracker.track(update: profileUpdate1)
+        appcues.storage.userID = "user-2" // mimick what is done in Appcues identify() call
+        tracker.track(update: profileUpdate2)
+
+        // Assert
+        waitForExpectations(timeout: 1)
+    }
+
+    func testIdentifyWithScreen() throws {
+        // Arrange
+        let onRequestExpectation = expectation(description: "Valid request")
+        let expectedEvents = [Event(name: "appcues:screen_view", attributes: ["screenTitle":"screen-name"])]
+
+        // In the case of a profile update, immediately followed by a group update, verify that the
+        // updates get merged together in a single Activity that gets sent to the ActivityProcessor
```

Comment:
I think this comment is a copy/paste error?

Reply:
yep! will clean up, thanks.